### PR TITLE
Holoparasite hosts can now be lings and/or have multiple parasites

### PR DESCRIPTION
**Pull request by Robustin**
Snowflake rules suck, trading a godly first life at the expense of being able to revive is a completely fair trade for 14TC+Ling status. 

Multiple parasites also aren't even that OP, just makes it harder to miss your lasers when killing the host. 

Edit: Will probably redo all this in another PR, but I'm tweaking Ling chems in this so assuage concerns about ling murderbone.

### CODE REVIEW

**Comment by Eaglendia** [comment]
You should PR these separately or Kor will come close this in like two minutes because multiple parasites don't match the lore

**Comment by GunHog** [comment]
This is a good PR. I like this PR.

**Comment by KorPhaeron** [comment]
I've already rejected both these individually.

**Comment by KorPhaeron** [comment]
You guys aren't the ones who will have to put up with being yelled at in OOC/deadchat/forums/IRC/github over parasites being OP when a ling with fleshmend lets its stand murder everyone. I don't want this headache.

**Comment by Fox-McCloud** [comment]
Just what we need, lings with holoparsites that are literally impossible to kill.

**Comment by KorPhaeron** [comment]
I also object to the premise of "the game gave you traitorling so its a godly life and every other player online just has to suck it up while you kill all of them with completely broken shit"

It's the purest form of Grief Lottery(TM) thinking

**Comment by Robustin** [comment]
Lings with HP's are easier to kill, you have twice as many targets to laser and once they are dead, they die forever.

I'd honestly consider it a nerf for 95% of the cases unless you get the rare robust ling + robust holoparasite. Losing revive is a big deal. 

This is a simple change Kor. Every week we completely fuck up some aspect of the game for 'test' purposes. I don't see why this shouldn't be tried out. You know in your heart that the vast majority of cases are going to be "idiot ling outs himself with newbie holoparasite and is murdered and turned to dust on the spot".

**Comment by PKPenguin321** [comment]
We should at least make holos refundable then

**Comment by Zilenan91** [comment]
Lings with sleeping carp are also kind of the same situation

**Comment by Robustin** [comment]
We're in this stupid limbo where most people seem to agree that putting all these snowflake restrictions on lings is bad, but coders managed to sneak in a few for holoparasites and now they'll reject any changes that would give us a consistency.

I fear a ling with an ebow and emag far more than one with a bonus target for me to laser or deny revives.

Multiple parasites is an even rarer case that shouldn't normally happen, and even still its not game-breaking at all given the amount of attention/damage you're going to attract splitting your health pool into 3 different valid targets.

**Comment by Robustin** [comment]
I'll even toss in a ling nerf real fast just so we can put this whole stupid "Once a month a good player goes on a rampage with a ling and despite the other 99/100 ending up as gibber food, that one good player sears its griefing into our brains and convinces us that lings need to be nerfed".

**Comment by Robustin** [comment]
Ugh. Need someone to merge #16164 so I can update my fork and address these conflicts. 

Kor why you always going for the nuclear option on this stuff? The chem changes to lings last year was a pretty drastic change, could've definitely been tweaked. But instead its either "REMOVE LINGS, REMOVE GLANDS" or we don't touch it at all. 

Lings were gimpy at 50/.5, but were pretty strong at 100/2. There's a middle ground for this shit that I'll address soon. Once I can address the conflicts.

**Comment by KorPhaeron** [comment]
>We're in this stupid limbo where most people seem to agree that putting all these snowflake restrictions on lings is bad

This is dumb.

We have a game with hundreds if not thousands of objects. It is unreasonable to expect or design around every single object when you have a new one in mind. When an unfair interaction crops up, we have 3 choices

1. Remove one of the two objects. This doesn't sound like much fun, does it? Coders get yelled at for nerfing/removing.

2. Let the objects interact, damn the consequences. People abuse the broken flavour of the month for the max bodycount. Coders are yelled at for not fixing it. Admins are yelled at. Everyone is salty.

3.  Stop the objects interacting. Apparently this is a grave offense because it's snowflake. I guess I still get yelled at.

As for carp/ling (which was another NO DON'T STOP THE INTERACTION WITH SNOWFLAKE that was shouted down).

I think my final bodycount for last nights sleeping carp+changeling was the whole station (who had actually organized to rush me all at once with a greyshirt militia+all of sec) +an ERT+(2?) more ERTs that died to environmental hazards trying to catch me in lavaland+drone deathsquad with gyrojets and flamethrowers+slaughter demon before a full on deathsquad finally got a lucky 1 hit knockdown with an esword. That doesn't sound like much fun to have on a regular basis, does it?

**Comment by KorPhaeron** [comment]
>But instead its either "REMOVE LINGS, REMOVE GLANDS" or we don't touch it at all.

We gave lings glands by fucking default, it's not a nuclear option to stop them taking it again.

And see above about killing fucking everyone.

**Comment by Robustin** [comment]
They were given glands because they were unplayable without them. It was a shitty design that basically forced lings to spend 2 points on glands or else be the most useless piece of shit antag on the planet. 

With default glands they were playable without taking glands, but the coder made the NEW glands much more powerful by scaling it all the same despite the base being doubled.

The smart solution was to go to a middle ground. Making the base storage/regen strong enough that you didn't feel like a piece of shit if you didn't take it, but also make glands feel like a useful option that didn't completely trivialize chem costs. The choice should have always been "take 10 points of powers for BREADTH and the cost of DEPTH" or "take 8 points of powers + engorged for DEPTH at the cost of BREADTH". 

That's easily doable with tweaking 2-4 variables. It's a trivial effort. Its all a sliding scale. 50/.5 was shit. 100/2 opened the door to some really OP ling players. Somewhere in the middle there's a good balance. I was shooting for 50/.8-1 and 75/1.2-1.5 engorged.... that's a fair balance that makes lings weaker than they've been since mid-2015,  without letting them spam fleshmend+adrenals, but also without making their abilities literally something where they get to cast one power once every minute while the entire station validhunts them.

**Comment by Supermichael777** [comment]
##### IF LINGS WITH TRAITOR SHIT IS OP MAKE BEING NATURALLY BOTH IMPOSSIBLE.

File: code/game/gamemodes/changeling/changeling.dm [hunk 1]
```diff
@@ -282,8 +282,8 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 	var/dna_max = 6 //How many extra DNA strands the changeling can store for transformation.
 	var/absorbedcount = 0
 	var/chem_charges = 20
-	var/chem_storage = 75
-	var/chem_recharge_rate = 1
+	var/chem_storage = 50
```
Comment (Iamgoofball) [inline code/game/gamemodes/changeling/changeling.dm]: uhhhh

File: code/game/gamemodes/changeling/powers/glands.dm [hunk 2]
```diff
@@ -9,5 +9,5 @@
 	..()
 	var/datum/changeling/changeling=user.mind.changeling
 	changeling.chem_storage += 25
-	changeling.chem_recharge_rate *=2
-	return
\ No newline at end of file
+	changeling.chem_recharge_rate *=1.5
```
Comment (Iamgoofball) [inline code/game/gamemodes/changeling/powers/glands.dm]: uhhhh

File: code/game/gamemodes/changeling/changeling.dm [hunk 3]
```diff
@@ -47,7 +47,7 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 
 /datum/game_mode/changeling/announce()
 	world << "<b>The current game mode is - Changeling!</b>"
-	world << "<b>There are alien changelings on the station. Do not let the changelings succeed!</b>"
+	world << "<b>There are alien changelings on the station.ch Do not let the changelings succeed!</b>"
```
Comment (Iamgoofball) [inline code/game/gamemodes/changeling/changeling.dm]: uhhhh.ch

File: code/game/gamemodes/changeling/powers/glands.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/game/gamemodes/changeling/powers/glands.dm]: Yeah what the fuck.

>nerfing ling overall just to make it slightly less OP when it gets traitorling

Comment (Iamgoofball) [inline code/game/gamemodes/changeling/powers/glands.dm]: undocumented changes much?

**Comment by Archie700** [comment]
Ok why are these changes not stated in the description

**Comment by duncathan** [comment]
Ch

File: code/game/gamemodes/changeling/powers/glands.dm (hunk 2 above)
Comment (Jordie0608) [inline code/game/gamemodes/changeling/powers/glands.dm]: https://github.com/tgstation/-tg-station/pull/16457#issuecomment-202592392
should be in the desc but they were mentioned

**Comment by PKPenguin321** [comment]
i appreciate the idea robustin but this is terribly executed

File: code/game/gamemodes/changeling/changeling.dm (hunk 3 above)
Comment (Core0verload) [inline code/game/gamemodes/changeling/changeling.dm]: ch?

File: code/game/gamemodes/changeling/changeling.dm [hunk 4]
```diff
@@ -282,8 +282,8 @@ var/list/slot2type = list("head" = /obj/item/clothing/head/changeling, "wear_mas
 	var/dna_max = 6 //How many extra DNA strands the changeling can store for transformation.
 	var/absorbedcount = 0
 	var/chem_charges = 20
-	var/chem_storage = 75
-	var/chem_recharge_rate = 1
+	var/chem_storage = 50
+	var/chem_recharge_rate = 0.8
```
Comment (Core0verload) [inline code/game/gamemodes/changeling/changeling.dm]: ~?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 5]
```diff
@@ -615,12 +615,6 @@
 
 /obj/item/weapon/guardiancreator/attack_self(mob/living/user)
 	for(var/mob/living/simple_animal/hostile/guardian/G in living_mob_list)
-		if (G.summoner == user)
-			user << "You already have a [mob_name]!"
```
Comment (Core0verload) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: You forgot to remove line 617.

**Comment by Robustin** [comment]
None of this is getting merged in this PR. I was just showing what the numbers would look like. Kor removed glands and my fork still has it, so I don't know how I can fix that. I'm hoping my Zombie PR gets merged soon so I can just start fresh. 

It's an overall nerf to ling IMO but it could be seen as a buff if you're the type of ling that relies primarily on fleshmend/adrenal and everything else if just fluff. 

Really like glands? Buff
Not a fan of glands? Nerf

As I posted before, glands had a purpose, and a bunch of short-sighted coding shouldn't destroy an otherwise sensible Ling power. 

Since when did killing the station become some fucking Holocaust-tier event that we need to bend over backwards to prevent? If a robust player gets two antags in one, they should always have a shot at wiping the crew out, especially in a gametype where the crew might be distracted with other threats (which according to the forums, is exactly what happened Kor). A revolver and autolathe is all a robust player has ever needed to completely annihilate the entire crew. Is Carp+Ling good? Yea its nice, but you're still sacrificing all ranged weaponry for 17TC. You know what else is good? Ling + any decent traitor loadout. Ebow + thermals or Ebow + emag or Ebow + noslips + soap is still a scarier loadout to me. WIth EMP + a ranged stun with infinite ammo + fleshmend/adrenals/armblade I'd struggle to come up with a reliable way to stop them. With a carp ling, it boils down to "don't let him melee you, use a stun weapon to melee him first".

**Comment by AlexanderUlanH** [comment]
Killing the station should be something that a sufficiently clever player manages to do once in a blue moon, requiring enough skill that the 20-30 dead people watching have to begrudgingly sit back and nod in appreciation. If it's easy or happens often it's just frustrating.
Honestly though I think balancing around traitorling is pointless. Better to remove the chance to be triator and ling and just allow both of those antag kits to work by themselves.

**Comment by KorPhaeron** [comment]
@AlexanderUlanH 

Said it better than I was, thanks

**Comment by Incomptinence** [comment]
Remove tatorling, its the real snowflake.

**Pull request closed without merging**